sequencer/src/l1_client.rs

```diff
@@ -121,26 +121,32 @@ impl L1Client {
 
     /// Get information about the given block.
     ///
-    /// If the desired block number is not available yet, this function will block until it becomes
-    /// available.
-    pub async fn wait_for_block(&self, number: u64) -> L1BlockInfo {
+    /// If the desired block number is not finalized yet, this function will block until it becomes
+    /// finalized.
+    pub async fn wait_for_finalized_block(&self, number: u64) -> L1BlockInfo {
         let interval = self.provider.get_interval();
 
-        // Wait for the block to become available.
-        loop {
-            match self.provider.get_block_number().await {
-                Ok(height) if height > number.into() => break,
-                Ok(height) => {
-                    tracing::warn!(number, %height, "waiting for L1 block");
-                }
-                Err(err) => {
-                    tracing::error!(%err, "failed to get L1 block height");
-                }
+        // Wait for the block to finalize.
+        let finalized = loop {
+            let Some(block) = self.get_finalized_block().await else {
```

Comment:
I think using our `get_finalized_block` proxy here is causing us to do extra work. It calls external `get_finalize_block` which internally calls `provider.get_block` and pulls the block_number and hash off of the block. So for this PR my suggestion is to just use `self.provider.get_block` here to reduce the number of HTTP requests. Long term I think it points to a need to refactor L1Client to better fit it different use cases.

Reply:
If we have the block in the first loop, can we not avoid the second loop entirely? This would avoid however many requests are made in the second loop. Which would probably just be 1 given that we've already shown that the block is present.

Reply:
Ah no now I understand we first need to find out what the finalized block is. I was misreading.